src/template-tags/tickets.php

```diff
@@ -174,119 +176,63 @@ function tribe_tickets_display_count_and_ticket_button() {
+				$cta_html .= '<span class="tribe-out-of-stock">' . esc_html_x( 'Sold out', 'list view stock sold out', 'event-tickets' ) . '</span>';
+			} else {
+				$stock = $data['stock'];
+				if (
+					$data['unlimited'] ||
+					( $data['count'] && ! $data['stock'] )
+				) {
+					// if unlimited tickets, tickets with no stock and rsvp, or no tickets and rsvp unlimited - hide the remaining count
+					$stock = false;
+				}
 
-			if (
-				$types['tickets']['unlimited']
-				||
-				(
-					$types['tickets']['count']
-					&& ! $types['tickets']['stock']
-					&& $types['rsvp']['count']
-				)
-				||
-				(
-					! $types['tickets']['count']
-					&& $types['rsvp']['unlimited']
-				)
-			) {
-				// if unlimited tickets, tickets with no stock and rsvp, or no tickets and rsvp unlimited - hide the remaining count
-				$stock = false;
-			}
+				$stock_html = '';
+				if ( $stock ) {
+					$number = number_format_i18n( $stock );
+					if ( 'rsvp' === $type ) {
+						$text = _n( '%s spot left', '%s spots left', $stock, 'event-tickets' );
+					} else {
+						$text = _n( '%s ticket left', '%s tickets left', $stock, 'event-tickets' );
+					}
+
+					$stock_html = '<span class="tribe-tickets-left">';
+					$stock_html .= esc_html( sprintf( $text, $number ) );
+					$stock_html .= '</span>';
+				}
 
-			$stock_html = '';
-			if ( $stock ) {
-				$stock_html = '<span class="tribe-tickets-left">';
-				if ( $rsvp ) {
-					$stock_html .= esc_html(
-						sprintf(
-							_n(
-								'%s spot left',
-								'%s spots left',
-								$stock,
-								'event-tickets'
-							),
-							number_format_i18n( $stock )
-						)
-					);
+				if ( 'rsvp' === $type ) {
+					$button_label  = esc_html_x( 'RSVP Now!', 'list view rsvp now ticket button', 'event-tickets' );
+					$button_anchor = '#rsvp-now';
 				} else {
-					$stock_html .= esc_html(
-						sprintf(
-							_n(
-								'%s ticket left',
-								'%s tickets left',
-								$stock,
-								'event-tickets'
-							),
-							number_format_i18n( $stock )
-						)
-					);
+					$button_label  = esc_html_x( 'Buy Now!', 'list view buy now ticket button', 'event-tickets' );
+					$button_anchor = '#buy-tickets';
 				}
-				$stock_html .= '</span>';
-			}
-
-			$button_label  = esc_html_x( 'Buy Now!', 'list view buy now ticket button', 'event-tickets' );
-			$button_anchor = '#buy-tickets';
-			if ( $rsvp ) {
-				$button_label  = esc_html_x( 'RSVP Now!', 'list view rsvp now ticket button', 'event-tickets' );
-				$button_anchor = '#rsvp-now';
-			}
 
-			$button = '<form method="get" action="' . esc_url( get_the_permalink( $event_id ) . $button_anchor ) . '"><button type="submit" name="tickets_process" class="tribe-button">' . $button_label . '</button></form>';
+				$button = '<form method="get" action="' . esc_url( get_the_permalink( $event_id ) . $button_anchor ) . '">';
+				$button .= '<button type="submit" name="tickets_process" class="tribe-button">' . $button_label . '</button>';
```

Comment:
I don't really care that much, but fwiw we could concat without repeating the use of `$button`, ie:

```php
$button = 'One, two,'
        . 'Buckle my shoe.';
```